packages/editor/src/components/editor-history/redo.js

```diff
@@ -24,7 +24,14 @@ export default compose( [
 	withSelect( ( select ) => ( {
 		hasRedo: select( 'core/editor' ).hasEditorRedo(),
 	} ) ),
-	withDispatch( ( dispatch ) => ( {
-		redo: () => dispatch( 'core/editor' ).redo(),
+	withDispatch( ( dispatch, ownProps ) => ( {
+		redo: () => {
+			// If there are no redo levels this is a no-op, because we don't actually
```

Comment:
In treating `EditorHistoryRedo` as the presentational component of the [Presentational / Container pair](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0), it seems like if it's meant to be a disabled behavior, it should be considered by the component in deciding to invoke the `onClick` callback.

Even something like:

```jsx
onClick={ hasRedo ? redo : undefined }
```

It's not so much a concern here in that the components are so tied to the specific editor use case of Undo/Redo, but is a good rule of thumb in delegating where logic should occur between the underlying component and its enhancer(s).

Reply:
Oh, good point. Follow-up PR with a fix is here: #11428.